qcodes/instrument/base.py

```diff
@@ -596,6 +596,8 @@ def __init__(
 
         self._t0 = time.time()
 
+        if not name.isidentifier():
+            raise ValueError(f"{name} invalid instrument identifier")
```

Comment:
I think it might make sense to put this in the InstrumentBase class since not only istruments but actually also the channels and modules should have names that are valid identifiers